coins/monero/src/wallet/seed/polyseed.rs

```diff
@@ -0,0 +1,419 @@
+    WordList { words, has_prefix, has_accent }
+  }
+}
+
+// language words
+const CS_WORDS: &str = include_str!("./polyseed/cs.json");
+const FR_WORDS: &str = include_str!("./polyseed/fr.json");
+const KO_WORDS: &str = include_str!("./polyseed/ko.json");
+const EN_WORDS: &str = include_str!("./polyseed/en.json");
+const IT_WORDS: &str = include_str!("./polyseed/it.json");
+const ES_WORDS: &str = include_str!("./polyseed/es.json");
+const JA_WORDS: &str = include_str!("./polyseed/ja.json");
+const PT_WORDS: &str = include_str!("./polyseed/pt.json");
+const ZH_S_WORDS: &str = include_str!("./polyseed/zh_simp.json");
+const ZH_T_WORDS: &str = include_str!("./polyseed/zh_trad.json");
+
+lazy_static! {
+  static ref LANGUAGES: HashMap<Language, WordList> = HashMap::from([
+    (Language::Czech, WordList::new(serde_json::from_str(CS_WORDS).unwrap(), true, false)),
+    (Language::French, WordList::new(serde_json::from_str(FR_WORDS).unwrap(), true, true)),
+    (Language::Korean, WordList::new(serde_json::from_str(KO_WORDS).unwrap(), false, false)),
+    (Language::English, WordList::new(serde_json::from_str(EN_WORDS).unwrap(), true, false)),
+    (Language::Italian, WordList::new(serde_json::from_str(IT_WORDS).unwrap(), true, false)),
+    (Language::Spanish, WordList::new(serde_json::from_str(ES_WORDS).unwrap(), true, true)),
+    (Language::Japanese, WordList::new(serde_json::from_str(JA_WORDS).unwrap(), false, false)),
+    (Language::Portuguese, WordList::new(serde_json::from_str(PT_WORDS).unwrap(), true, false)),
+    (Language::ChineseSimp, WordList::new(serde_json::from_str(ZH_S_WORDS).unwrap(), false, false)),
+    (Language::ChineseTrad, WordList::new(serde_json::from_str(ZH_T_WORDS).unwrap(), false, false)),
+  ]);
+}
+
+pub struct PolySeedData {
+  pub birthday: u64,
+  pub features: u32,
+  pub entropy: Zeroizing<[u8; 32]>,
+  pub checksum: u16,
+}
+
+impl PolySeedData {
+  /// creates a `PolySeedData` with current time birthday, random entropy and no features.
+  pub fn new<R: RngCore + CryptoRng>(rng: &mut R) -> PolySeedData {
+    // get birthday
+    let time = SystemTime::now().duration_since(UNIX_EPOCH).unwrap().as_secs();
+    let birthday = birthday_encode(time);
+
+    // get the entropy
+    let mut key = random_scalar(rng).to_bytes();
+    key[(SECRET_SIZE + 1) ..].fill(0);
+    key[SECRET_SIZE - 1] &= CLEAR_MASK;
+    let entropy = Zeroizing::new(key);
+
+    // make polyseed data
+    let mut data = PolySeedData { birthday, features: 0, entropy, checksum: 0 };
+
+    // set the  checksum
+    data.checksum = create_checksum(&data);
+
+    data
+  }
+
+  pub fn from(data: PolySeedData) -> Option<PolySeedData> {
+    // check features
+    let features = make_features(data.features);
+    if !polyseed_features_supported(features) {
+      return None;
+    }
+
+    // get birthday
+    let birthday = birthday_encode(data.birthday);
+
+    // make sure it is a valid scalar
+    Scalar::from_canonical_bytes(*data.entropy)?;
```

Comment:
... why is this here? Anything which was created with

```rs
    let mut key = random_scalar(rng).to_bytes();
    key[(SECRET_SIZE + 1) ..].fill(0);
    key[SECRET_SIZE - 1] &= CLEAR_MASK;
    let entropy = Zeroizing::new(key);
```

will match this (as that's a 19-byte value needing to fit into a 31.5-byte space).

The concern I have is the non-19-byte values fitting into that 31.5 space.